Makefile

```diff
@@ -339,7 +339,7 @@ clean: ## Clean up resources created by make targets
 	rm -rf ./bin/*
 	rm -rf ./pkg/executables/cluster-name/
 	rm -rf ./pkg/providers/vsphere/test/
-	find . -depth -type d -regextype posix-egrep -regex '.*\/Test.*-[0-9]{9}\/.*' -exec rm -rf {} \;
+	find -E . -depth -type d -regex '.*\/Test.*-[0-9]{9}\/.*' -exec rm -rf {} \;
```

Comment:
why is this change necessary?

Reply:
This is a incompatible find command parameter between linux and macos, my version works for macos. The existed version does not work in my mac. 

Can you also do a test to see if it happens on your mac? You can run make clean to trigger that command.

Reply:
Gotcha, the old version didn't work for me either on Mac, nice catch

Reply:
Updated with conditional statement to work for both MacOs and Linux